internal/xds/translator/testdata/out/xds-ir/credential-injection.listeners.yaml

```diff
@@ -0,0 +1,62 @@
+- address:
+    socketAddress:
+      address: 0.0.0.0
+      portValue: 10080
+  defaultFilterChain:
+    filters:
+    - name: envoy.filters.network.http_connection_manager
+      typedConfig:
+        '@type': type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+        commonHttpProtocolOptions:
+          headersWithUnderscoresAction: REJECT_REQUEST
+        http2ProtocolOptions:
+          initialConnectionWindowSize: 1048576
+          initialStreamWindowSize: 65536
+          maxConcurrentStreams: 100
+        httpFilters:
+        - disabled: true
+          name: envoy.filters.http.credential_injector/httproute/default/httproute-1/rule/0/match/0/gateway_envoyproxy_io
+          typedConfig:
+            '@type': type.googleapis.com/envoy.extensions.filters.http.credential_injector.v3.CredentialInjector
+            credential:
+              name: envoy.http.injected_credentials.generic
+              typedConfig:
+                '@type': type.googleapis.com/envoy.extensions.http.injected_credentials.generic.v3.Generic
+                credential:
+                  name: credential_injector/credential/httproute/default/httproute-1/rule/0/match/0/gateway_envoyproxy_io
+                  sdsConfig:
+                    ads: {}
+                    resourceApiVersion: V3
+        - disabled: true
+          name: envoy.filters.http.credential_injector/httproute/default/httproute-1/rule/1/match/0/gateway_envoyproxy_io
```

Comment:
something for the future, should this be renamed to be unique to a httpRouteFilter, similar to how we create a filter per policy ?

Reply:
Good catch! Injecting multiple credentials is not a common case, but still possible.

Reply:
Good catch!